models/issue.go

```diff
@@ -1348,22 +1348,50 @@ func (opts *IssuesOptions) setupSessionNoLimit(sess *xorm.Session) {
 	}
 
 	if opts.User != nil {
-		sess.And(
-			issuePullAccessibleRepoCond("issue.repo_id", opts.User.ID, opts.Org, opts.Team, opts.IsPull.IsTrue()),
-		)
+		repoCond, err := issuePullAccessibleRepoCond("issue.repo_id", opts.User.ID, opts.Org, opts.Team, opts.IsPull.IsTrue())
+		if err != nil {
+			return err
+		}
+		sess.And(repoCond)
 	}
+	return nil
 }
 
 // issuePullAccessibleRepoCond userID must not be zero, this condition require join repository table
-func issuePullAccessibleRepoCond(repoIDstr string, userID int64, org *organization.Organization, team *organization.Team, isPull bool) builder.Cond {
+func issuePullAccessibleRepoCond(repoIDstr string, userID int64, org *organization.Organization, team *organization.Team, isPull bool) (builder.Cond, error) {
 	cond := builder.NewCond()
 	unitType := unit.TypeIssues
 	if isPull {
 		unitType = unit.TypePullRequests
 	}
 	if org != nil {
 		if team != nil {
-			cond = cond.And(teamUnitsRepoCond(repoIDstr, userID, org.ID, team.ID, unitType)) // special team member repos
+			// If the current user is a admin, it doesn't necesarly mean
+			// it has joined the team, but we still should return all repo's
+			// of that team.
+			isAdmin, err := org.IsOwnedBy(userID)
+			if err != nil {
+				return nil, err
+			}
+			if isAdmin {
+				cond = cond.And(builder.In(repoIDstr,
```

Comment:
I personally don't like if this cause needs to be extracted as I feel this could easily be misused(this doesn't have the check if the user is in the team)